pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -209,42 +211,86 @@ func (c *Controller) removeFinalizer(pvc *v1.PersistentVolumeClaim) error {
 }
 
 func (c *Controller) isBeingUsed(pvc *v1.PersistentVolumeClaim) (bool, error) {
+	// Look for a Pod using pvc in the Informer's cache. If one is found the
+	// correct decision to keep pvc is taken without doing an expensive live
+	// list.
+	if inUse, err := c.askInformer(pvc); err != nil {
+		// No need to return because a live list will follow.
+		klog.Error(err)
+	} else if inUse {
+		return true, nil
+	}
+
+	// Even if no Pod using pvc was found in the Informer's cache it doesn't
+	// mean such a Pod doesn't exist: it might just not be in the cache yet. To
+	// be 100% confident that it is safe to delete pvc make sure no Pod is using
+	// it among those returned by a live list.
+	return c.askAPIServer(pvc)
```

Comment:
We may need evaluate the load increase since this func may be frequently called when when we delete pvcs.

Reply:
There is a correct way to reduce that load.  See the "A correct optimization" section in https://www.ibm.com/cloud/blog/new-builders/preserve-kubernetes-api-objects-while-they-are-in-use?mhsrc=ibmsearch_a&mhq=preserve%20kubernetes%20api%20object

Reply:
That optimization is already part of this PR.

Reply:
Sorry, did not read enough.